### DISCUSSION TITLE
What are these unit tests trying to tell me?

### DISCUSSION THREAD

**Discussion by patrickmaynard** [symfony/symfony#47729] [Q&A]
I finally got back to issue #46867 and have a pull request open. 

https://github.com/symfony/symfony/issues/46867

I unfortunately am getting a couple of failing unit tests, and I'm not entirely sure I understand what they're trying to tell me. Here is the run:

https://github.com/symfony/symfony/runs/8100289308?check_suite_focus=true

... and here is the diff showing what was changed:

https://github.com/symfony/symfony/pull/47438/files

Any ideas about what I need to change in my branch in order to make the tests work?

**Comment by stof** [comment]
Well, you removed code dealing with NoSuchIndexException in an array in `getPropertyValue`, which is probably what causes those tests to fail.

**Reply by patrickmaynard** [reply]
Ah, yes, on line 82/86. I see it now. Thank you!

**Reply by patrickmaynard** [reply]
Nope, that wasn't it. I've added that logic back into the `PropertyPathAccessor` class, and I'm still getting the same failure. 

Which is strange because I now am literally just catching a specific type of exception, then adding a bit of extra information to the message and rethrowing it. There are now no other changes in my PR.

https://github.com/symfony/symfony/pull/47438/files

Very odd.
